### PR TITLE
fix: Adding service monitor for retina-operator

### PR DESCRIPTION
**Pull request by mereta**
# Description

Adding a ServiceMonitor for retina-operator
* parameterized & applied retina-operator name
* adding service & serviceMonitor CRD's for retina-operator
* applied appropriate relabeling & metric relabeling config to align with retina-jobs additional scrape config 

## Related Issue

retina-operator wasn't being scraped for metrics by prometheus.
Initially it was appearing in the 'retina-pods' job and failing as reported in this issue:
#738 

Partial fix was merged to remove the operator pod from the list here:
https://github.com/microsoft/retina/pull/770

## Checklist

- [x] I have read the [contributing documentation](https://retina.sh/docs/contributing).
- [x] I signed and signed-off the commits (`git commit -S -s ...`). See [this documentation](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification) on signing commits.
- [x] I have correctly attributed the author(s) of the code.
- [x] I have tested the changes locally.
- [x] I have followed the project's style guidelines.
- [x] I have updated the documentation, if necessary.
- [x] I have added tests, if applicable.

## Screenshots (if applicable) or Testing Completed

image

Operator specific metrics with job='retina-operator' selector:

image

## Additional Notes

Proposed next steps is to align the way we add scrap configs:
#847

---

Please refer to the [CONTRIBUTING.md](../CONTRIBUTING.md) file for more information on how to contribute to this project.

### CODE REVIEW

**Comment by mereta** [comment]
@microsoft-github-policy-service agree company="Microsoft"

File: deploy/legacy/manifests/controller/helm/retina/values.yaml [hunk 1]
```diff
@@ -253,7 +263,20 @@ metrics:
     tlsConfig: {}
     ## @param metrics.serviceMonitor.relabelings [array] Prometheus relabeling rules to apply to samples before scraping
     ##
-    relabelings: []
+    relabelings:
```
Comment (SRodi) [inline deploy/legacy/manifests/controller/helm/retina/values.yaml:266]: I guess you added the config here as the relabeling is going to be shared between the operator and agent in future work part of #847 right?

Comment (mereta) [reply deploy/legacy/manifests/controller/helm/retina/values.yaml:266]: Yes exactly, then all service monitors would be aligned with what we currently have in the additionalScrapeConfig.

**Review by SRodi — COMMENTED** [review]
this LGTM @mereta - can we also push the chart update for hubble control plane deploy as part of this PR?

**Comment by mereta** [comment]
> this LGTM @mereta - can we also push the chart update for hubble control plane deploy as part of this PR?

Working on it! ✅

**Comment by mereta** [comment]
> this LGTM @mereta - can we also push the chart update for hubble control plane deploy as part of this PR?

@SRodi I will hold off on the Hubble charts. I ran in to a few issues with the operator, I cant seem to get any metrics form /metrics endpoint. It seems to be different to the legacy operator. I reached out to @rbtr to have a look and maybe advide.
Maybe we create a separate issue for that.

**Review by SRodi — DISMISSED** [review]

**Comment by SRodi** [comment]
@mereta it looks like E2E tests need to be updated as part of this PR - see this failure https://github.com/microsoft/retina/actions/runs/11331610343/job/31512301186#step:6:213

```sh
    runner.go:27: 
        	Error Trace:	/home/runner/work/retina/retina/test/e2e/framework/types/runner.go:27
        	            				/home/runner/work/retina/retina/test/e2e/retina_e2e_test.go:77
        	Error:      	Received unexpected error:
        	            	did not expect error from step InstallHelmChart but got error: failed to install chart: unable to build kubernetes objects from release manifest: resource mapping not found for name: "retina-operator-servicemonitor" namespace: "kube-system" from "": no matches for kind "ServiceMonitor" in version "monitoring.coreos.com/v1"
        	            	ensure CRDs are installed first
        	Test:       	TestE2ERetina
```

**Comment by mereta** [comment]
> @mereta it looks like E2E tests need to be updated as part of this PR - see this failure https://github.com/microsoft/retina/actions/runs/11331610343/job/31512301186#step:6:213
> 
> ```shell
>     runner.go:27: 
>         	Error Trace:	/home/runner/work/retina/retina/test/e2e/framework/types/runner.go:27
>         	            				/home/runner/work/retina/retina/test/e2e/retina_e2e_test.go:77
>         	Error:      	Received unexpected error:
>         	            	did not expect error from step InstallHelmChart but got error: failed to install chart: unable to build kubernetes objects from release manifest: resource mapping not found for name: "retina-operator-servicemonitor" namespace: "kube-system" from "": no matches for kind "ServiceMonitor" in version "monitoring.coreos.com/v1"
>         	            	ensure CRDs are installed first
>         	Test:       	TestE2ERetina
> ```

I'm going to create a separate PR for this. The ServiceMonitor CRD is not available in the cluster, as prometheus is not installed.
For SeriviceMonitor to be the default way to collect metrics we need to address this issue by pre-installing the CRD in the e2e cluster. This is a pre-req for #847

**Review by SRodi — APPROVED** [review]
Thanks @mereta! Just to clarify for everyone: retina-operator `ServiceMonitor` is currently disabled as we need to update E2E tests to install Prometheus ServiceMonitor CRDs - that will be captured in separate GH issue and tackled in another PR.

https://github.com/microsoft/retina/blob/3009f2864b8d42b5f30d8b8e10323e096d10fc28/deploy/legacy/manifests/controller/helm/retina/values.yaml#L228

**Pull request merged**